### ISSUE TITLE
no-unused-variable: falsely reported within curried functions (FP)

### ISSUE THREAD

**Issue by miguelnavarro89** [palantir/tslint#3312]
### Bug Report

- tslint@5.7.0
- typescript@2.4.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
...
const count: number = length(requiredKeys);
    const messageArticle: string = ifElse(
      multiple,
      always('Los campos'),
      always('El campo')
    )(count);
...
```

Terminal output:
```terminal
'count' is declared but never used. (no-unused-variable)
  70 |     const requiredKeys: string[] = keys(required);
  71 |
> 72 |     const count: number = length(requiredKeys);
     |          ^
  73 |     return ifElse(
  74 |       hasAny,
  75 |       always(
```

with `tslint.json` configuration:

```json
...
 "no-unused-expression": true,
 "no-unused-variable": true,
...

```

#### Actual behavior
Reported as `no-unused-variable`

#### Expected behavior
Should pass correctly
